src/averageAges.js

```diff
@@ -55,10 +79,22 @@
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const filteredPeople = people.filter((person) =>
+    onlyWithSon
+      ? people.some(
+        (mother) => mother.name === person.mother && person.sex === 'm'
+      )
+      : people.find((mother) => mother.name === person.mother)
+  );
+  
+  const ageDifferences = filteredPeople.map(
+    (person) =>
+      person.born - people.find((mother) => mother.name === person.mother).born
```

Comment:
The 'find' method is used inside a 'map' method which will increase time complexity. Consider finding all mothers before mapping over 'filteredPeople'.